klarna-order-management-for-woocommerce.php

```diff
@@ -292,37 +292,60 @@ public function cancel_klarna_order( $order_id, $action = false ) {
 		 * @param array $items Order items.
 		 * @param bool  $action If this was triggered by an action.
 		 *
-		 * @return bool|WP_Error Returns bool true if updating was successful or a WP_Error object if not.
+		 * @return WP_Error|true Returns true if updating was successful or a WP_Error object if not.
 		 */
 		public function update_klarna_order_items( $order_id, $items, $action = false ) {
 			$options = self::get_instance()->settings->get_settings( $order_id );
-			if ( ! isset( $options['kom_auto_update'] ) || 'yes' === $options['kom_auto_update'] || $action ) {
+			$order   = wc_get_order( $order_id );
 
-				$order = wc_get_order( $order_id );
+			if ( ! in_array( $order->get_payment_method(), array( 'klarna_payments', 'kco' ), true ) ) {
+				return new \WP_Error( 'not_klarna_order', 'Order does not have klarna_payments or kco payment method.' );
+			}
+
+			// Are we on the subscription page?
+			if ( 'shop_subscription' === $order->get_type() ) {
+				$token_key = 'klarna_payments' === $order->get_payment_method() ? KP_Subscription::RECURRING_TOKEN : '_kco_recurring_token';
```

Comment:
Move the check for the payment method into the if statement to prevent issues with potentially trying to continue even after we know we should not. So for example:
```php
if ( 'shop_subscription' === $order->get_type() 
    && in_array( $order->get_payment_method(), array( 'klarna_payments', 'kco' ), true ) ) {
```

Reply:
We check for the payment method on line 301. The control on 307 is for determining the name of the recurring token.